routes/bundles.js

```diff
@@ -0,0 +1,28 @@
+// bundles.js
+//
+// Serve up some sweet Browserify bundles
+//
+// Copyright 2018 AJ Jordan <alex@strugee.net>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+"use strict";
+
+var browserify = require("browserify-middleware"),
+    path = require("path");
+
+var addRoutes = function(app) {
+    app.get("/javascript/pump.js", browserify(path.join(__dirname, "../public/javascript/pump.js")));
```

Comment:
I think is good idea minified and make a bundle, but is not a good idea for performance make this from the server, you can make the bundle from CLI  in build scripts, also we don's use the commonJs on the client side (dependencies, imports etc) so Browserify adds unnecessary extra code for that.

If the idea is to convert the client side to a commonJs structure I think webpack os more powerful because you can make [lazyLoad](https://webpack.js.org/guides/lazy-loading/) and other advantages.

Also, you can make the bundle only with [UglifyJS2](https://github.com/mishoo/UglifyJS2) from CLI without extra code and add some watch task for the development process.

Also, I think is better keep independent the frontEnd process like create bundles, for that reason I closed https://github.com/pump-io/pump.io/pull/1350 also for performance reasons and is similar to [browserify-middleware](https://github.com/ForbesLindesay/browserify-middleware)

Reply:
> also we don's use the commonJs on the client side (dependencies, imports etc) so Browserify adds unnecessary extra code for that

Right, the plan was to switch to using CommonJS since it just makes it so much easier to structure the code. Really not even "easier", but "possible at all".

Re: WebPack, I thought about it, but I like Browserify better for its flexibility and I tend to agree with [this assessment by @substack (the author)](https://gist.github.com/substack/68f8d502be42d5cd4942#browserify-for-webpack-users):

> I think that longer-term, separability has more benefits from a maintenance and experimentation perspective, but that doesn't mean that you can't do pretty much everything webpack can do in the browserify ecosystem. It just means that there might be several different ways of doing that task to serve different use-cases. Diversity is good!. Diversity means that you can choose an approach that more closely matches the trade-offs that you need, but you might need to put in more work to evaluate the different options.

Also WebPack is famous for having complicated, hard-to-debug/patch/etc. config files.

> Also, I think is better keep independent the frontEnd process like create bundles, for that reason I closed #1350 also for performance reasons and is similar to browserify-middleware

That's funny, since that was opened I actually have changed my mind on it and I think caching in memory is probably better, both for contributor experience and for production. Serving from memory is faster and you don't have to rebuild in development. FWIW that's what Express does with compiled templates.